Добавлен диалог "График работ'''

Soldatkina-Alina opened the pull request:
- добавлены таблицы в БД
- созданы классы для работы с графиком работ и диалог
- добавлена возможность сохранять график работы в БД

Soldatkina-Alina committed:
Добавлена проверка на обязательный ввод телефона

Soldatkina-Alina committed:
В БД добавлены новые таблицы

- сотрудники
- график работы
- связующая таблица между рабочим днем и назначенным на него сотрудником

Soldatkina-Alina committed:
Добавлены новые классы и их связь с БД

- класс сотрудников
- класс графика работ
- класс сотрудников для одного рабочего дня

Soldatkina-Alina committed:
Добавлен диалог "График работ" с возможностью сохранения графика в БД

-добавлено только сохранение в БД, без просмотра и редактирования на форме
-добавлены сопутствующие изменения в другие классы

Gankov commented on CarGlass/Dialogs/SheduleDlg.cs:
```diff
@@ -0,0 +1,230 @@
+		private void Fill()
+		{
+			var sql = "select shw.id, emp.id, emp.first_name, emp.last_name, emp.patronymic from CarGlass.employees emp " +
+				" left join CarGlass.shedule_employee_works shew on shew.id_employee = emp.id" +
```
Вот это вообще в корне не верно, нельзя в коде указывать название базы. Тогда не возможо будет при необходимости заставить работать с другой базой.

Gankov commented on CarGlass/Dialogs/SheduleDlg.cs:
```diff
@@ -0,0 +1,230 @@
+		{
+			foreach(object[] emp in EmployeeWorkList)
+			{
+				emp[2] = emp[2].ToString().Replace("-", "");
```
Я не очень понял зачем здесь вот это? какая разница как они будут отображаться в виде - или нет. На сохранение разве это влияет?

Gankov commented on CarGlass/Dialogs/SheduleDlg.cs:
```diff
@@ -0,0 +1,230 @@
+
+		void CellText_Edited2(object o, EditedArgs args)
+		{
+			TreeIter iter;
```
Совсем не понял зачем здесь возможно редактировать имя сотрудника? Они же не будут каждый день имя писать вручную? Зачем тогда таблица сотрудников нужна?

Gankov replied:
Я честно говоря предпологал что будет справочник сотрудников, где они забьют всех своих 5 сотрудников. один раз напишут и имя. А каждый день им будут просто ставить галочки вышел на смену или нет.
Я если честно твоей идеи не понимаю. А как им дальше ЗП при таком походе считать будешь?
Вернее я вообще не понимаю, твоего решения. Когда я буду редактировать имя сотрудника я поменяю его и на всех остальных днях или только для этого? Если для этого. тогда как ты будешь потом за неделю за месяц ЗП им показывать, по текстовым полям сравнивать что они одинаково написаны? А если сегодня одного и того же написали так, а завтра иначе. У них в ЗП вместо 5, 10 человек будет? Потом на калькуляторе объединять в одного?

Soldatkina-Alina committed:
Добавлены функции просмотра, редактирования и удаления для графика работ

Soldatkina-Alina committed:
Исправлен баг при открытии меню

Soldatkina-Alina committed:
Добавлены изменения в график сотрудников для лучшего отображения

Soldatkina-Alina committed:
Изменена БД

- добавлены некоторые незначительные изменения в новые таблицы справочников и таблицу связи услуг и исполнителей

Soldatkina-Alina committed:
Добавлены новые классы для справочников

- и к ним виджеты и mapping

Soldatkina-Alina committed:
В окно заказа добавлены исполнители

Gankov commented on CarGlass/Dialogs/AddEditFormulas.cs:
```diff
@@ -0,0 +1,96 @@
+	{
+		IUnitOfWork UoW = UnitOfWorkFactory.CreateWithoutRoot();
+		Domain.SalaryFormulas salaryFormulas = null;
+		Coefficients coefficients = null;
```
Зачем эта переменная нужна глобально?

Gankov replied:
И более того она вроде даже локально не нужна, нигде не используется.

Gankov commented on CarGlass/Dialogs/AddEditFormulas.cs:
```diff
@@ -0,0 +1,96 @@
+			this.Build();
+			this.salaryFormulas = salaryFormulas;
+			if(salaryFormulas.Id != 0)
+				listSalaryFormulas = UoW.Session.QueryOver<Domain.SalaryFormulas>(() => salaryFormulas).Where(x => x.Id == salaryFormulas.Id).List();
```
Возможно это место работает, но тут явно путаницу вызывает salaryFormulas и приходит, в виде аргумента метода, и указывается как алиас, в запросе, и далее в Where, совершенно не очевидно, оно сработает как алиас или как аргумет полученный в метод. Назови пожалуйста алиас и переменную поразному чтобы не было путаницы что из них что.

Gankov commented on CarGlass/Dialogs/AddEditFormulas.cs:
```diff
@@ -0,0 +1,96 @@
+		{
+			Coefficients coeff = ytreeCoeff.GetSelectedObject<Coefficients>();
+			listCoefficients.Remove(coeff);
+			ytreeCoeff.ItemsDataSource = listCoefficients;
```
Тут сейчас и испрвлять наверно не надо, но на будущее если использовать Observabe коллекции то такого делать не придется.

Gankov commented on CarGlass/Dialogs/AddEditFormulas.cs:
```diff
@@ -0,0 +1,96 @@
+
+		protected void OnButtonCancelClicked(object sender, EventArgs e)
+		{
+			this.Destroy();
```
Здесь не правильно. нужн не разрушать окно а таким же образом ответить Respond(ResponseType.Cancel); Например. Потому что внешний код у тебя наверняка ждет ответа, и должен будет после ответа, разрушить окно сам.

Gankov commented on CarGlass/Dialogs/EmployeesCatalog.cs:
```diff
@@ -0,0 +1,142 @@
+namespace CarGlass.Dialogs
+{
+	[System.ComponentModel.ToolboxItem(true)]
+	public partial class EmployeesCatalog : Gtk.Bin
```
Зря ты не стала придерживаться соглашения, называть все диалоги с окончанием Dlg, чтобы не было пересечения с другими классами. Сейчас уже запаришся исправлять так что не меняй.

Gankov commented on CarGlass/Dialogs/EmployeesCatalog.cs:
```diff
@@ -0,0 +1,142 @@
+		private void createTable()
+		{
+			ytree.ItemsDataSource = listEmployees.Where(x => x.StatusEmployee.Code == ActiveCode).ToList();
+			ytree.ColumnsConfig = ColumnsConfigFactory.Create<Employee>()
```
А зачем заново все время табличку то перенастраивать? Ничего же не меняется.

Gankov commented on CarGlass/Dialogs/EmployeesCatalog.cs:
```diff
@@ -0,0 +1,142 @@
+			listEmployees.Remove(delEmp);
+			foreach(var emp in listEmployees)
+				UoW.Save(emp);
+			UoW.Commit();
```
Тут лучше комит делать в конце, потому что комит сохраняет все изменения в сессии. И так как ты не все сохранила, он может ругруться в какой то момен что есть ссылки на что-то не сохраненое, поэтому не буду делать комит.

Gankov commented on CarGlass/Dialogs/OrderDlg.cs:
```diff
@@ -24,7 +24,11 @@ public partial class OrderDlg : FakeTDIEntityGtkDialogBase<WorkOrder>
 				typeof(string), // 2 - Наименование
 				typeof(double), // 3 - Цена
 				typeof(long)    // 4 - order_pay id
+				,typeof(bool)
```
А комментарий хот можно было прописать что, за что отвечает.

Gankov commented on CarGlass/Dialogs/SalarySettingsDlg.cs:
```diff
@@ -0,0 +1,17 @@
+
+namespace CarGlass.Dialogs
+{
+	public partial class SalarySettingsDlg : Gtk.Dialog
```
Это мусор или действительно надо?

Gankov commented on CarGlass/Domain/Coefficients.cs:
```diff
@@ -0,0 +1,41 @@
+		{
+		}
+
+		public Coefficients(string name, string comment)
```
На будущее такие вещи проще сделать со значениями по умолчанию string comment = null
Тогда можно будет сделать всего 1 конструктор а не 2.

Soldatkina-Alina committed:
Добавлены проверки при сохранении формул и коэффициентов

Soldatkina-Alina committed:
Добавлена проверка на удаление коэффициента

Soldatkina-Alina committed:
Добавлены заголовки и исправлены мелкие ошибки

Soldatkina-Alina committed:
Добавлен расчет ЗП

Soldatkina-Alina committed:
Добавлены дополнительные проверки
